src/Agent.Worker/Build/BuildDirectoryManager.cs

```diff
@@ -510,6 +519,23 @@ private BuildCleanOption GetBuildDirectoryCleanOption(IExecutionContext executio
                 }
             }
         }
+
+        private string GetDefaultRepositoryPath(
+            IExecutionContext executionContext,
+            RepositoryResource repository,
+            string defaultSourcesDirectory)
+        {
+            if (RepositoryUtil.HasMultipleCheckouts(executionContext.JobSettings))
+            {
+                // If we have multiple checkouts they should all be rooted to the work directory
+                return Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Work), RepositoryUtil.GetCloneDirectory(repository));
```

Comment:
This is incorrect. I need to add the build directory into this. I.e. _work/1/reponame